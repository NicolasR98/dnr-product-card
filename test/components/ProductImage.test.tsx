import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/products';

describe('Tests on ProductImage', () => {
    test('Should display correctly', () => {
        const wrapper = renderer.create(
            <ProductImage img='image.jpg'/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('should display correctly with the product image', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})
