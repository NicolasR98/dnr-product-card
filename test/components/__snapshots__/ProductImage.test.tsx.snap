// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on ProductImage Should display correctly 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="image.jpg"
/>
`;

exports[`Tests on ProductImage should display correctly with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
